City-of-Helsinki/drupal-palvelukeskus
# UHF-8539 Platform config v3

# [UHF-8539](https://helsinkisolutionoffice.atlassian.net/browse/UHF-8539)


## What was done


* Updated `hdbt_subtheme` based on platform config v3 changes
* Updated `hdbt_subtheme` tools
* Updated `helfi_platform_config` to v3 and all configurations based on new base config

## How to install and test

* Read the platform v3 update documentation
   * https://github.com/City-of-Helsinki/drupal-helfi-platform-config/blob/3.x/documentation/update.md#update-instructions

* Make sure your instance is up and running on dev branch
  * `git checkout main`
  * `make fresh`
* Run `make drush-cr`
* Chekout this branch 
   * `git checkout UHF-8539_platform_config_v3`
* Run `composer install`
* Run `make shell`
   * Run `drush cr`, `drush helfi:platform-config:update-database` and re-import configuration `drush cim -y`
   * If the configuration import fails, run `drush cr` and `drush cim -y` until the import is successful
* There should not be any visible changes to the site. Check that the site works as before.

## Notice!
* When running the update in test / prod, this needs to be run separately as it will update the banner designs
`drush eval "module_load_include('install', 'helfi_paragraphs_banner', 'helfi_paragraphs_banner'); helfi_paragraphs_banner_update_9001();"`

[UHF-8539]: https://helsinkisolutionoffice.atlassian.net/browse/UHF-8539?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
